splunk_add_on_ucc_framework/ucc_ui_lib/src/main/webapp/components/DeleteModal.jsx

```diff
@@ -0,0 +1,137 @@
+    width: 800px
+`;
+
+class DeleteModal extends Component {
+    static contextType=InputRowContext;
+
+    constructor(props){
+        super(props);
+        this.state = {isDeleting:false,ErrorMsg:""};
+    }
+    
+    handleRequestClose = () => {
+        this.props.handleRequestClose();
+    };
+
+    parseErrorMsg = (msg) => {
+        let errorMsg = ''; let regex; let matches;
+        try {
+            regex = /.+"REST Error \[[\d]+\]:\s+.+\s+--\s+([\s\S]*)"\.\s*See splunkd\.log(\/python.log)? for more details\./;
+            matches = regex.exec(msg);
+            if (matches && matches[1]) {
+                try {
+                    const innerMsgJSON = JSON.parse(matches[1]);
+                    errorMsg = String(innerMsgJSON.messages[0].text);
+                } catch (error) {
+                    // eslint-disable-next-line prefer-destructuring
+                    errorMsg = matches[1];
+                }
+            } else {
+                errorMsg = msg;
+            }
+        } catch (err) {
+            errorMsg = 'Error in processing the request';
+        }
+        return errorMsg;
+    }
+
+    handleDelete = () => {
+        this.setState( (prevState)=> {
+            return {...prevState, isDeleting:true,ErrorMsg:""}
+        }, ()=>{
+
+            axiosCallWrapper({
+                serviceName: `${this.props.serviceName}/${this.props.stanzaName}`,
+                customHeaders: { 'Content-Type': 'application/x-www-form-urlencoded' },
+                method: 'delete',
+                handleError: false
+            }).catch((err) => {
+                const errorSubmitMsg= this.parseErrorMsg(err?.response?.data?.messages[0]?.text);
+                this.setState({ErrorMsg:errorSubmitMsg,isDeleting:false});
+                return Promise.reject(err);
+
+            }).then((response) => {
+                
+                this.context.setRowData( update(this.context.rowData,{[this.props.serviceName]: {$unset : [this.props.stanzaName]}}))
+                this.setState({isDeleting:false});
+                this.handleRequestClose()
+                
+            });
+        });
+
+        
+    };
+
+    // Display error message 
+    generateErrorMessage = () => {
+            if (this.state.ErrorMsg) {
+                return (
+                    <div>
+                        <Message appearance="fill" type="error">
+                            {this.state.ErrorMsg}
+                        </Message>
+                    </div>
+                )
+            }
+            return null; 
+        }
+
+    render() {
+        let deleteMsg;
+        if(this.props.isInput){
+            deleteMsg = `Are you sure you want to delete "${this.props.stanzaName}" ?`;
+        }
+        else{
+            deleteMsg = `Are you sure you want to delete "${this.props.stanzaName}" ? Ensure that no input is configured with "${this.props.stanzaName}" as this will stop data collection for that input.`;
+        }
```

Comment:
Here we can assign a value while declaring a variable. like:

> let deleteMsg = `Are you sure you want to delete "${this.props.stanzaName}" ?`;

and then simply we can check if **!this.props.isInput** then append another string to deleteMsg var

Reply:
It will loose readability.